mounts.go

```diff
@@ -35,6 +40,9 @@ type ContainerMountSource interface {
 	// Type determines the final mount type
 	// possible options are limited by the Docker API
 	Type() MountType
+
+	// Validate validates the source of the mount
+	Validate() error
```

Comment:
I was forced to add this to have early feedback when preparing the mounts for a container

Reply:
suggestion: make is an optional interface so we don't have no-op implementations all over.

This would mean we get the following in `preCreateContainerHook`:
```go
	var mountErrors []error
	for _, m := range req.Mounts {
		if v, ok := m.source.(Validater); ok {
			if err := m.Source.Validate(); err != nil {
				mountErrors = append(mountErrors, err)
			}
		}
	}

	if len(mountErrors) > 0 {
		return errors.Join(mountErrors...)
	}
```